### ISSUE TITLE
Issue with the "mailto" anchors

### ISSUE THREAD

**Issue by couteaudepoche** [skinofthesoul/grav-plugin-antispam#27]
Antispam plugin v1.5.7 and Page Toc plugin v3.2.4 activated : Issue with the "mailto" anchors 

The Antispam code in the HTML send to the browser looks like:
document.write("\<a href='mailto:"+link+"' >"+link+"");

However, Antispam injects the following code into the page before it is being send to the browser:
document.write("\<a href='mailto:"+link+"' >"+link+"<\/a>");

---------------------------------------------------------------------------
A bug has been logged on [PHP: Bug #74628  DOM auto remove HTML closing tag in <script> tag when save .](https://bugs.php.net/bug.php?id=74628)
It describes exactly the issue of DOMDocument "eating" closing tags in javascript.

The thread offers two different solutions:

1. Add a backslash (\\) at the closing: <\/a> tag becomes <\/a>
This should be done by the author of AntiSpam in antispam.php line 196
Add an extra flag LIBXML_SCHEMA_CREATE when loading HTML into the DOMDocument
2. This should be done by the author of PageToc in file HtmlHelper.php on line 37

**Comment by skinofthesoul** [comment]
Hey there,
I’m super sorry I am only getting back to this now. I’m guessing this is still a problem?

I will try to reproduce the issue in a bit. If it really is only a matter of putting in a backslash this should be fixed super quickly, but as we all know it might break something else. I will get back to you later!

**Comment by skinofthesoul** [comment]
Ok actually I just realised I have an existing website using both Antispam v1.5.7 and Page Toc v3.2.4, and there does not seem to be a problem with either the mailto links nor the page toc links. I can’t find a removed closing mailto tag either. Could you tell me more about what is not working for you?

**Comment by pamtbaau** [comment]
Yes, it still happens. Unfortunately OP didn't really create a sensible issue...

Try the following:
- Using latest default Grav install
- Install antispam
- Install page-toc
- Add to page Typography the following lines
  ```
  ul>
  <li><a href="mailto:qq@ww.ee">My email address</a></li>
  <li>next line</li>
  </ul>
  ```
- Open Typography in the browser. You will see:
<img width="291" height="89" alt="Image" src="https://github.com/user-attachments/assets/10316d65-9782-4f0b-abf5-0c39e650dc27" />
  
  As you can see, the "next line" has become part of the email link.
- Disable page-toc
- Refresh browser. You will see:
<img width="333" height="106" alt="Image" src="https://github.com/user-attachments/assets/b7208cab-b41b-4b3e-9820-8a76059aab7b" />

  "next line" is no longer part of the email link.

The problem is caused by page-toc, because it uses `loadHTML()` from lib libxml, which eats away the closing `</a>`. See https://bugs.php.net/bug.php?id=74628. The bug description offers 2 solutions:

1.
> [2017-05-22 16:37 UTC] [cmb@php.net](https://people.php.net/cmb)
Unfortunately, libxml isn't really HTML5 compatible yet, and there
isn't much we can do about that. You have to stick to conforming
HTML 4.01, and write <\/h1>, see <https://3v4l.org/ldGmr>.

This means adding a backslash to the closing tag of the email in the antispam plugin.
2.
> [2021-03-04 11:34 UTC] gijovarghese141 at gmail dot com
Adding LIBXML_SCHEMA_CREATE to loadHTML() as an option will fix the issue

$dom->loadHTML($html, LIBXML_SCHEMA_CREATE);

I think using suggestion 2. in your page-toc would be best.

**Comment by skinofthesoul** [comment]
Wow, thank you for the detailed reply!! I will take some time to check this out and add a fix soonish (two to three weeks).

**Comment by pamtbaau** [comment]
Sorry, I thought page-toc was your plugin... [Trilby media](https://github.com/trilbymedia/grav-plugin-page-toc) should fix its page-toc. It can be resolved in your anti-spam, but that is a lame workaround.

**Comment by skinofthesoul** [comment]
Thank you for the detailed research!! In this case I am marking this as closed here.
